vip-scanner/checks/VIPRestrictedVariablesCheck.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+class VIPRestrictedVariablesCheck extends BaseCheck
+{
+	function check( $files ) {
+		$result = true;
+
+		$checks = array(
+			'((?<![\\\'\"])\$\$[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*?(?![\\\'\"])|(?<![\\\'\"])\$\{(?:.*)[\}](?![\\\'\"]))' => array( "level" => "Warning", "note" => "Possible variable variables" ),
```

Comment:
> How can we reduce the possibility of false positives in single quoted strings?

The regex currently uses a set of negative lookarounds matching for single and double quotes.  If they are found around the variable, the match is not made.

`(?<![\\\'\"])` ... `(?![\\\'\"])`